cnd/src/connector.rs

```diff
@@ -0,0 +1,128 @@
+use crate::{
+    network::{DialInformation, Network},
+    seed::Seed,
+    swap_protocols::{
+        self,
+        asset::Asset,
+        metadata_store::{self, InMemoryMetadataStore, MetadataStore},
+        rfc003::{
+            self,
+            alice::{RequestError, SendRequest},
+            state_machine::SwapStates,
+            state_store::{self, InMemoryStateStore, StateStore},
+            ActorState, CreateLedgerEvents, Ledger,
+        },
+        LedgerConnectors, Metadata, SwapId,
+    },
+};
+use futures::{sync::oneshot::Sender, Future};
+use libp2p::PeerId;
+use libp2p_comit::frame::Response;
+use std::sync::Arc;
+
+/// Connector is used to connect incoming messages from the HTTP API with logic
+/// that triggers outgoing messages on the libp2p layer.
+#[derive(Debug)]
```

Comment:
:wrench: I think none of this will be useful to print for debugging purposes, so I would just use `#[allow(missing_debug_implementations)]`. Alternatively, you can use the already imported [`rust-derivative`](https://github.com/mcarton/rust-derivative) crate to selectively ignore fields when deriving `Debug`.